app/services/link_reporter_csv_service.rb

```diff
@@ -0,0 +1,70 @@
+class LinkReporterCsvService
+  delegate :<<, :close, to: :csv
+
+  def initialize(reports_dir:, organisation:)
+    @organisation = organisation
+    @reports_dir = reports_dir
+  end
+
+  def generate
+    csv = CSV.open(file_path, "w", encoding: "UTF-8")
+    csv << headings
+    public_editions.each { |edition| add_edition_row(csv, edition) }
+  ensure
+    csv.close
+  end
+
+private
+
+  attr_reader :organisation, :reports_dir
+
+  def public_editions
+    Edition.publicly_visible.with_translations.in_organisation(organisation)
+  end
+
+  def file_path
+    reports_dir.join("#{@organisation.slug}_links_report.csv")
+  end
+
+  def headings
+    ["page", "admin link", "public timestamp", "format", "broken link count", "broken links"]
+  end
+
+  def add_edition_row(csv, edition)
+    csv_row_for_edition = [
+      public_url(edition),
+      admin_url(edition),
+      timestamp(edition),
+      edition.type,
+      broken_links(edition).size,
+      broken_links(edition).join("\r\n")
+    ]
+    csv << csv_row_for_edition
+  end
+
+  def public_url(edition)
+    Whitehall.url_maker.public_document_url(edition, host: public_host, protocol: "https")
+  end
+
+  def admin_url(edition)
+    Whitehall.url_maker.admin_edition_url(edition, host: admin_host, protocol: "https")
+  end
+
+  def timestamp(edition)
+    edition.public_timestamp.to_s
+  end
+
+  def broken_links(edition)
+    edition.link_check_reports.last.broken_links.map(&:uri)
+  end
+
+  # These hosts are hardcoded because we run this on integration but want the
```

Comment:
I think it's better that we get integration URLs on integration, staging URLs on staging, and then production URLs on production.